### DISCUSSION TITLE
Adding a function to pass to `vim.ui.input`

### DISCUSSION THREAD

**Discussion by jandamm** [MunifTanjim/nui.nvim#50] [Ideas]
From what I've seen this plugin is a component library.

Would you be interested in increase the scope and add functions which can be used directly or can be used as a template for others?
Out of my head I'd think of 
- rename for lsp (basically what you've posted in your [blog post](https://muniftanjim.dev/blog/neovim-build-ui-using-nui-nvim))
- function to pass to `vim.ui.select`
- function to pass to `vim.ui.input`

**Comment by MunifTanjim** [comment]
Hey! Thanks for checking out nui.nvim.

Those are all very good use-cases for nui! 😃

> Would you be interested in increase the scope and add functions which can be used directly or can be used as a template for others?

Hmm, I think we should refrain from doing that in the main library.

`nui.nvim` is very general purpose and is intended to be used by other plugins. If we start to include codes for applications of the components, it might get out of hands... 😂 you know, because you can do a whole lot of things with these components (lsp helpers, diagnostic helpers, file explorer and what-nots...)

But [nui.nvim Wiki](https://github.com/MunifTanjim/nui.nvim/wiki) could be a good place to document examples for these various use-cases 😃

> function to pass to vim.ui.select

In case you want to take a look at an example for this:

https://github.com/MunifTanjim/dotfiles/blob/f207d4c3/private_dot_config/nvim/lua/config/lsp/custom.lua#L70-L128

**Reply by jandamm** [reply]
Yeah, that's kinda the reason I opened a discussion instead of an issue.

I'll have a look at your implementation and check if I get it to work for input. If I do, I'll share it here 👍

**Reply by MunifTanjim** [reply]
This looks like something you'd be interested in: https://github.com/stevearc/dressing.nvim

**Reply by jandamm** [reply]
Thanks for the hint, I'll definitely try this one out 👍

**Reply by MunifTanjim** [reply]
Added that Wiki page: https://github.com/MunifTanjim/nui.nvim/wiki/vim.ui

**Reply by jandamm** [reply]
Awesome thank you ☺️

**Reply by nghialm269** [reply]
hi @MunifTanjim, the `vim.ui.input` example in the wiki makes the cursor jump back one character after confirming, which make renaming variable of one character impossible, is there anyway to fix this?

**Reply by MunifTanjim** [reply]
@nghialm269 can you check if the latest commit https://github.com/MunifTanjim/nui.nvim/commit/6f803e88093573f73d4ee6c0dfe0575df3f97a9f fixes it?

**Reply by nghialm269** [reply]
@MunifTanjim Thanks, it works perfectly now!
